src/modules/collection/sagas.ts

```diff
@@ -538,6 +540,31 @@ export function* collectionSaga(builder: BuilderAPI, catalyst: CatalystClient) {
     return allItems.filter(item => item.collectionId === collection.id)
   }
 
+  function* updateItemCurationsStatus(itemCurations: ItemCuration[], status: CurationStatus) {
+    const effects: CallEffect<ItemCuration>[] = itemCurations.map(curation => {
+      return call([builder, 'updateItemCurationStatus'], curation.itemId, status)
+    })
+    const newItemCuration: ItemCuration[] = yield all(effects)
+    return newItemCuration
+  }
+
+  function* getItemsAndEntitiesToDeploy(collection: Collection, tree?: MerkleDistributorInfo, hashes?: Record<string, string>) {
```

Comment:
Ok so, this works! but I think it's a miss-leading signature for the method. The problem is not your implementation but the fact that we decided to treat `TP items` the same as `standard Items`. But anyways here we are.

What I mean by miss-leading is that, if I look at this method without *a lot of context*, I can't really tell that the last two parameters mean "it's TP".

```ts
getItemsAndEntitiesToDeploy(standardCollection)

getItemsAndEntitiesToDeploy(tpCollection, tree, hashes)
```

nothing tells me that I can't make this call: `getItemsAndEntitiesToDeploy(tpCollection, tree, hashes)`.
Same thing for `buildItemEntity`.

SO, I was thinking how to fix this aaaand... I can't really come up with a better answer than making three methods. A generic one used internally and two public ones *OR* an override (but I'll be the first time we do it)

```ts
function* getItemsAndEntitiesToDeploy(collection: Collection) {
function* getItemsAndEntitiesToDeploy(collection: Collection,  tree: MerkleDistributorInfo, hashes: Record<string, string>) {

function* getStandardItemsAndEntitiesToDeploy(collection: Collection) {
function* getTPtemsAndEntitiesToDeploy(collection: Collection,  tree: MerkleDistributorInfo, hashes: Record<string, string>) {

// same idea for buildItemEntity
```

What do you think? I'd like to know what @LautaroPetaccio thinks about this too tho

Reply:
Updated to something similar to what you suggested. It's more code but it's more readable.

Reply:
I think that re-using this function will be a problem, specially because:
- We don't need the item entities for the TP items, as the syncing of the TP items can be checked by using the item's `content_hash` and `local_content_hash`
- The `getApprovalData` endpoint was created to not need all the items in the UI, but for the standard items, that is required, as it needs to show them. We're going to show only an amount of them.